### PR TITLE
include subfolders feature

### PR DESCRIPTION
**Pull request by MisterY**
## Connection with issue(s)

Resolve issue #429 

## Testing and Review Notes

I added the setting to the Experimental features.
Please do provide some feedback. I'm still learning!

### CODE REVIEW

**Review by vHanda — COMMENTED** [review]
In general it looks great. Lets merge it after I relicense the files. However, I wouldn't mark the bug fixed with just this.

Ideally, when searching in a folder which has sub-folders we should show the user a way to control the search. I have some ideas of a rough UI, but I think we both will need to think about this. The simplest would to just show a toggle bar -

```
------------------------------
| search word                |
------------------------------
| Search in subfolder    Yes/no|
------------------------------
| results                    |
|                            |
------------------------------
 ```
 
 Maybe in the future we can make this much fancier when we want to expose searching by case, if it has a link, lists, etc.

File: lib/screens/folder_listing.dart [hunk 1]
```diff
@@ -19,18 +22,29 @@ class FolderListingScreen extends StatefulWidget {
 class _FolderListingScreenState extends State<FolderListingScreen> {
   final _folderTreeViewKey = GlobalKey<FolderTreeViewState>();
   NotesFolderFS selectedFolder;
+  AppSettings settings;
```
[the remark this answers is not in the document]
Comment (MisterY) [reply lib/screens/folder_listing.dart]: You mean like below?

**Comment by MisterY** [comment]
> much fancier when we want to expose searching by case, if it has a link, lists,

I fully agree. Initially I thought a simple toggle would work but didn't want to add it yet to the production UI and, in the end, just went with the experimental switch for now.

File: lib/screens/folder_listing.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (vHanda) [reply lib/screens/folder_listing.dart]: Yes. You can remove this line. And below just add `var` before the `settings = Provider ...`

File: lib/screens/folder_listing.dart [hunk 2]
```diff
@@ -22,14 +21,13 @@ class FolderListingScreen extends StatefulWidget {
...
   NotesFolderFS selectedFolder;
-  AppSettings settings;
 
   @override
   Widget build(BuildContext context) {
     final notesFolder = Provider.of<NotesFolderFS>(context);
 
     // Load experimental setting
-    settings = Provider.of<AppSettings>(context);
+    var settings = Provider.of<AppSettings>(context);
```
Comment (vHanda) [inline lib/screens/folder_listing.dart:30]: perfect!

**Comment by vHanda** [comment]
Please rebase this :)

**Comment by MisterY** [comment]
k, done! thanks!

**Pull request merged**